sgl-kernel/include/utils.h

```diff
@@ -116,12 +129,37 @@ __device__ __forceinline__ float atomicMaxFloat(float* addr, float value) {
   return old;
 }
 
+__device__ __forceinline__ float warpReduceSum(float sum_value) {
+  sum_value += SGLANG_SHFL_XOR_SYNC(0xffffffff, sum_value, 16);
+  sum_value += SGLANG_SHFL_XOR_SYNC(0xffffffff, sum_value, 8);
+  sum_value += SGLANG_SHFL_XOR_SYNC(0xffffffff, sum_value, 4);
+  sum_value += SGLANG_SHFL_XOR_SYNC(0xffffffff, sum_value, 2);
+  sum_value += SGLANG_SHFL_XOR_SYNC(0xffffffff, sum_value, 1);
+  return sum_value;
+}
+
+__device__ __forceinline__ float blockReduceSum(float sum_value) {
```

Comment:
@hebiao064 Probably can help double confirm this one. (though also not being used)